### ISSUE TITLE
Bypass Paywall and CAPTCHA

### ISSUE THREAD

**Issue by waybackarchiver** [wabarc/wayback#144]
Launch **headfull browser** with [Xvfb](https://en.wikipedia.org/wiki/Xvfb) and import extensions.

Extensions:

- [Buster](https://github.com/dessant/buster)
- [Privacy Pass](https://github.com/privacypass/challenge-bypass-extension)
- [bypass-paywall](https://github.com/iamadamdev/bypass-paywalls-chrome)
- [uBlock](https://github.com/gorhill/uBlock)
- [Cookie Popup Blocker](https://github.com/n4cr/cookiepopupblocker)
- ...

Relates to #92

**Comment by waybackarchiver** [comment]
> dump the CDP messages and generate go code

This approach may complicate and add uncertainty to the situation.

> Provide a node runtime in go side or the browser side

As expected, this approach provides a new extension to call the methods exposed by 'puppeteer-extra-*'.

**Comment by hellodword** [comment]
> > dump the CDP messages and generate go code
> 
> This approach may complicate and add uncertainty to the situation.

Right, such as https://github.com/kkoooqq/fakebrowser/blob/586e85c0ed872513d2e0703d8c516250a8a4365b/src/core/DeviceDescriptor.ts#L463-L479

**Comment by hellodword** [comment]
> > Provide a node runtime in go side or the browser side
> 
> As expected, this approach provides a new extension to call the methods exposed by `puppeteer-extra-*`.

But this is complicate too, ESM and CJS, js and ts, dependencies, and so on. It sounds like a webpack in browser.

**Comment by waybackarchiver** [comment]
If the method on callsite or by extracting [`puppeteer-extra`](https://github.com/berstend/puppeteer-extra) does not work, we can search for alternative extensions or create one.

I'm working for launching Chrome and loading extensions. Next, make it possible to customize it so that it can load more extensions.

**Comment by waybackarchiver** [comment]
Related project [wabarc/starter](/wabarc/starter), and more details see [runs](https://github.com/wabarc/starter/runs/5439281973?check_suite_focus=true#step:5:81)

**Comment by waybackarchiver** [comment]
Relates to [wabarc/screenshot#11](https://github.com/wabarc/screenshot/issues/11)

**Comment by waybackarchiver** [comment]
The extension [bypass-paywall](https://github.com/iamadamdev/bypass-paywalls-chrome) is currently supported in conjunction with the [on-heroku](https://github.com/wabarc/on-heroku/tree/xvfb) project, and the next step will be to make the [starter](/wabarc/starter) extra approachable and to add more extensions to it.

Unfortunately, the incapability to save PDFs and screenshots in the X11 environment has arisen, which means that core idea may not be completely operational.
